src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.Value.cs

```diff
@@ -109,6 +107,26 @@ public IAbstractAnalysisValue GetNegatedValue()
                 return false;
             }
 
+            public static bool TryDecode(ImmutableArray<AttributeData> attributes, ArrayBuilder<PlatformMethodValue> infosBuilder)
+            {
+                foreach (var attribute in attributes)
+                {
+                    if (attribute.AttributeClass.Name is SupportedOSPlatformGuardAttribute or UnsupportedOSPlatformGuardAttribute &&
+                        HasNonEmptyStringArgument(attribute, out var argument) &&
+                        TryParsePlatformNameAndVersion(argument, out string platformName, out Version? version))
+                    {
+                        if (platformName.Equals(OSX, StringComparison.OrdinalIgnoreCase))
+                        {
+                            platformName = macOS;
+                        }
```

Comment:
I recommend moving this into `TryParsePlatformNameAndVersion` or the `PlatformMethodValue` constructor. We should limit the number of places this logic exists.